pkgs/by-name/go/got/package.nix

```diff
@@ -45,13 +46,9 @@ stdenv.mkDerivation (finalAttrs: {
     "-include getopt.h"
   ]);
 
-  doInstallCheck = true;
-
-  installCheckPhase = ''
-    runHook preInstallCheck
-    test "$($out/bin/got --version)" = "got ${finalAttrs.version}"
-    runHook postInstallCheck
-  '';
+  passthru.tests.version = testers.testVersion {
+    package = finalAttrs.finalPackage;
+  };
```

Comment:
@AndersonTorres, is this change a proper and correct improvement? I have yet to fully understand `passthru` and `testers.testVersion`, but from my current understanding this should be good. I'd truly appreciate your feedback and expertise on this.

Reply:
Yes, it is fine.

As I have explained before, `installCheckPhase` is meant to run checks provided by the upstream package developer. The difference between `checkPhase` and `installCheckPhase` is that `installCheckPhase` runs after installation while `checkPhase` runs after build.

Custom scripts are better suited to `passthru.tests`.

Reply:
Thanks for the explanation, @AndersonTorres, that's helpful!

Yet for some odd reason my 🧠 refuses to understand `passthru` (might be something to do with the wording that seems odd to me—not that it is, just that it makes it harder for me to grok it). Seems I need to revisit and re-read the [available documentation](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#var-stdenv-passthru)…

Reply:
`passthru` is similar to `meta` in the sense that a modification in `passthru` does not trigger a mass rebuild.

Usually we use `passthru` in order to collect some useful info from the derivation and expose it to the external world.

E.G. suppose you have a parameter, `uiType`, that enables a fancy UI in `got`, and the end user wants to know which GUI type was used (say, `cli, curses, gtk3, qt10`).

Then you can write

```nix
passthru = {
   uiType =
     if enableCurses then "curses"
     else if enableGtk3 then "gtk3"
     else "cli";
};
```

This way, the end user can evaluate `got.uiType` to know what UI was enabled.

------------------------------

Obviously the code above is a crude example. Usually we do cleaner and more useful stuff.

E.G. in `live555` we put a test: 

```nix
passthru.tests = {
   inherit vlc;
};
```

This way we can test this by calling `live555.tests.vlc`.

The idea is that `vlc` depends on `live555`, and sometimes the `vlc` team does not update its code when `live555` introduces a breaking change. So we at Nixpkgs catch this by running this test.